service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java

```diff
@@ -77,55 +87,79 @@
                 }
             } else {
                 return new ResponseEntity<>(parser.parse(RemremGenerateServiceConstants.NO_SERVICE_ERROR),
-                                            HttpStatus.SERVICE_UNAVAILABLE);
+                        HttpStatus.SERVICE_UNAVAILABLE);
             }
         } catch (Exception e) {
             return new ResponseEntity<>(null, HttpStatus.INTERNAL_SERVER_ERROR);
         }
     }
 
     /**
-     * this method is used to display the versions of generate and all loaded protocols.
+     * Used to display the versions of generate and all loaded protocols.
      *
-     * @return this method returns a json with version details.
+     * @return json with version details.
      */
+    @ApiOperation(value = "To get versions of generate and all loaded protocols", response = String.class)
     @RequestMapping(value = "/versions", method = RequestMethod.GET)
     public JsonElement getVersions() {
         Map<String, Map<String, String>> versions = new VersionService().getMessagingVersions();
         return parser.parse(versions.toString());
     }
-    
+
     /**
-     * this method returns available Eiffel event types as listed in EiffelEventType enum.
+     * Returns available Eiffel event types as listed in EiffelEventType enum.
      *
      * @return string collection with event types.
      */
+    @ApiOperation(value = "To get available eiffel event types based on the message protocol", response = String.class)
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "Event  types got successfully"),
+            @ApiResponse(code = 500, message = "Internal server error"),
+            @ApiResponse(code = 503, message = "Message protocol is invalid") })
     @RequestMapping(value = "/event_types/{mp}", method = RequestMethod.GET)
-    public ResponseEntity<Collection<String>> getEventTypes(@PathVariable("mp") String mp) {
-    	MsgService msgService = getMessageService(mp);    	
-    	return new ResponseEntity<Collection<String>>(msgService.getSupportedEventTypes(), HttpStatus.OK);
+    public ResponseEntity<?> getEventTypes(@ApiParam(value = "message protocol") @PathVariable("mp") String msgProtocol,
```

Comment:
missed required

Reply:
required = true is default

Reply:
you are right. for ApiParam default is reqiured = false